frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/openai/widgets/smart_edit_node_widget.dart

```diff
@@ -298,7 +299,26 @@ class _SmartEditInputState extends State<_SmartEditInput> {
       selection,
       texts,
     );
-    return widget.editorState.apply(transaction);
+    widget.editorState.apply(transaction).then(
+      (_) {
+        int endOffset = texts.last.length;
+        if (texts.length == 1) {
+          endOffset += selection.start.offset;
+        }
+        widget.editorState.updateCursorSelection(
+          Selection(
+            start: Position(
+              path: selection.start.path,
+              offset: selection.start.offset,
+            ),
+            end: Position(
+              path: [selection.start.path.first + texts.length - 1],
+              offset: endOffset,
+            ),
+          ),
+        );
+      },
+    );
```

Comment:
Is there a reason for using `.then` opposed to just `await` ? You're not relying on the result?
```suggestion
    await widget.editorState.apply(transaction);

    int endOffset = texts.last.length;
    if (texts.length == 1) {
      endOffset += selection.start.offset;
    }

    widget.editorState.updateCursorSelection(
      Selection(
        start: selection.start,
        end: Position(
          path: [selection.start.path.first + texts.length - 1],
          offset: endOffset,
        ),
       ),
    );
```